Invalid value 'null': expected a DateTime

When adding a filter `date: not empty` or `date: is empty` metabase returns an exception
```
Invalid value 'null': expected a DateTime.
```
---
![screenshot](http://get.antoine.io/0y1G1r160h2E/Image%202016-02-02%20at%2011.47.49%20AM.png)

---
I use an H2 database locally and connected Metabase to an AWS Redshift instance for querying.

## Metabase server logs:

```
2016-02-02 10:51:56,176 [ERROR] metabase.driver :: Query failure: Invalid value 'null': expected a DateTime.
2016-02-02 10:51:56,177 [DEBUG] metabase.models.interface :: DB CALL:  QueryExecution 34
2016-02-02 10:51:56,178 [DEBUG] metabase.middleware :: POST /api/dataset 200 (31 ms)
2016-02-02 10:53:54,173 [DEBUG] metabase.driver.query-processor ::
QUERY: ?
{:database 2,
 :type "query",
 :query
 {:source_table 7,
  :aggregation ["rows"],
  :breakout [],
  :filter ["AND" ["IS_NULL" 158]]},
 :constraints {:max-results 10000, :max-results-bare-rows 2000}}

2016-02-02 10:53:54,173 [WARN ] metabase.driver.query-processor.expand :: Specifying :rows as the aggregation type is deprecated in MBQL '98. This is the default behavior, so you don't need to specify it.
2016-02-02 10:53:54,173 [WARN ] metabase.driver.query-processor.expand :: Referring to fields by their bare ID (158) is deprecated in MBQL '98. Please use [:field-id 158] instead.
2016-02-02 10:53:54,173 [WARN ] metabase.driver.query-processor.expand :: Referring to fields by their bare ID (158) is deprecated in MBQL '98. Please use [:field-id 158] instead.
2016-02-02 10:53:54,173 [WARN ] metabase.driver.query-processor.expand :: You shouldn't specify an :and filter with only one subclause.
2016-02-02 10:53:54,174 [DEBUG] metabase.db.internal :: DB CALL:  Field [:description :table_id :special_type :name :parent_id :id :preview_display :display_name :base_type] (where (select* ENTITY21026) {:id [in field-ids]})
2016-02-02 10:53:54,174 [WARN ] metabase.driver.query-processor.expand :: Specifying :rows as the aggregation type is deprecated in MBQL '98. This is the default behavior, so you don't need to specify it.
2016-02-02 10:53:54,174 [WARN ] metabase.driver.query-processor.expand :: Referring to fields by their bare ID (158) is deprecated in MBQL '98. Please use [:field-id 158] instead.
2016-02-02 10:53:54,174 [WARN ] metabase.driver.query-processor.expand :: Referring to fields by their bare ID (158) is deprecated in MBQL '98. Please use [:field-id 158] instead.
2016-02-02 10:53:54,175 [WARN ] metabase.driver.query-processor.expand :: You shouldn't specify an :and filter with only one subclause.
2016-02-02 10:53:54,175 [DEBUG] metabase.db.internal :: DB CALL:  Field [:description :table_id :special_type :name :parent_id :id :preview_display :display_name :base_type] (where (select* ENTITY21026) {:id [in field-ids]})
2016-02-02 10:53:54,197 [ERROR] metabase.driver :: {:status :failed,
 :class java.lang.Exception,
 :error "Invalid value 'null': expected a DateTime.",
 :stacktrace
 ["metabase.driver.query_processor.parse$fn__20922.invoke(parse.clj:28)"
  "metabase.driver.query_processor.parse$fn__20907$G__20902__20914.invoke(parse.clj:9)"
  "metabase.driver.query_processor.resolve$value_ph_resolve_field.invoke(resolve.clj:118)"
  "metabase.driver.query_processor.resolve$fn__20949$G__20931__20956.invoke(resolve.clj:35)"
  "metabase.driver.query_processor.resolve$resolve_fields$fn__21034.invoke(resolve.clj:164)"
  "metabase.driver.query_processor.resolve$resolve_fields.invoke(resolve.clj:164)"
  "metabase.driver.query_processor.resolve$resolve.invoke(resolve.clj:219)"
  "metabase.driver.query_processor$pre_expand$fn__21161.invoke(query_processor.clj:117)"
  "metabase.driver.query_processor$wrap_catch_exceptions$fn__21156.invoke(query_processor.clj:100)"
  "metabase.driver.query_processor$process.invoke(query_processor.clj:330)"
  "metabase.driver$process_query.invoke(driver.clj:379)"
  "metabase.driver$dataset_query.invoke(driver.clj:419)"
  "metabase.api.dataset$fn__21288$fn__21291.invoke(dataset.clj:27)"
  "metabase.api.common.internal$wrap_catch_api_exceptions.invoke(internal.clj:216)"
  "metabase.api.dataset$fn__21288.invoke(dataset.clj:20)"
  "metabase.middleware$enforce_authentication$fn__23070.invoke(middleware.clj:82)"
  "metabase.api.routes$fn__23176.invoke(routes.clj:40)"
  "metabase.routes$fn__23746.invoke(routes.clj:23)"
  "metabase.middleware$log_api_call$fn__23163.invoke(middleware.clj:256)"
  "metabase.middleware$add_security_headers$fn__23123.invoke(middleware.clj:193)"
  "metabase.middleware$bind_current_user$fn__23074.invoke(middleware.clj:95)"
  "metabase.middleware$wrap_current_user_id$fn__23048.invoke(middleware.clj:64)"
  "metabase.middleware$wrap_api_key$fn__23089.invoke(middleware.clj:106)"
  "metabase.middleware$wrap_session_id$fn__23041.invoke(middleware.clj:55)"],
 :query
 {:type "query",
  :query
  {:source_table 7,
   :aggregation ["rows"],
   :breakout [],
   :filter ["AND" ["IS_NULL" 158]]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000}},
 :expanded-query nil}

2016-02-02 10:53:54,197 [ERROR] metabase.driver :: Query failure: Invalid value 'null': expected a DateTime.
2016-02-02 10:53:54,197 [DEBUG] metabase.models.interface :: DB CALL:  QueryExecution 35
2016-02-02 10:53:54,198 [DEBUG] metabase.middleware :: POST /api/dataset 200 (31 ms)
```

--- a/src/metabase/driver/query_processor/parse.clj
+++ b/src/metabase/driver/query_processor/parse.clj
@@ -24,5 +24,8 @@
       (instance? RelativeDatetime value)
       (map->RelativeDateTimeValue {:field this, :amount (:amount value), :unit (:unit value)})
 
+      (nil? value)
+      nil
+
       :else
       (throw (Exception. (format "Invalid value '%s': expected a DateTime." value))))))

